README.md

```diff
@@ -4,37 +4,105 @@
 
 The main objective of the Lab Results Adaptor is to hide complex legacy standards and instead present a simple and consistent interface aligned to current NHSD national standards. The adaptor removes the requirement for a GP System to handle the complexities of EDIFACT and MESH messaging.
 
-At a high level, the Lab Results Adaptor exposes a queue from which the GP System can consume medical reports
+At a high level, the Lab Results Adaptor exposes a queue (AMQP 1.0) from which the GP System can consume Pathology and Screening medical reports in FHIR DSTU3 format.
 
 See the [Resources](#resources) section for links to the underlying services and standards. 
 
-Adaptor handles two types of messages
-
+Adaptor handles following types of EDIFACT messages:
 - Pathology `NHS003`
 - Screening `NHS004`
 
-## Adaptor API
+and replies with `NHS001` (NHSACK) if requested
+
+## Architecture
+
+![Adaptor Component Diagram](/documentation/lab_results_diagram.png)
+![Adaptor Sequence Diagram](/documentation/lab_results_sequence_diagram.png)
+
+Adaptor downloads EDIFACT messages from MESH mailbox (`PATH_MEDRPT_V3` and `SCRN_BCS_MEDRPT_V4` MESH workflows) using MESH Client component and puts them on an internal `Inbound MESH Queue`. The Message Translator component fetches messages off the queue, translates the EDIFACT into FHIR format and puts the result onto the `Outbound GP Queue` where other GP Supplier apps can consume medical reports from.
+In the end, if EDIFACT Interchange Header requested such, an NHSACK confirmation is sent back to the labtoraty MESH mailbox.
+
+### Performance
+
+Adaptor's EDIFACT to FHIR translation steps are built for high performance and resiliency by applying horizontal scaling technique. However, the MESH consumer step performance cannot be increased this way. Due to the nature of MESH, if there are multiple adaptor instances, only one randomly chosen will download messages and put them on the `Inbound MESH Queue`.
+
+### Message processing and error handling
+
+When an EDIFACT message is downloaded from MESH using the MESH Client component, it's structure is not verified at this step. The message is stored on the internal `Inbound MESH Queue` and MESH is instructed that the file can be deleted since the message has been successfully downloaded.
```

Comment:
```suggestion
When an EDIFACT message is downloaded from MESH using the MESH Client component, its structure is not verified at this stage. The message is stored on the internal `Inbound MESH Queue` and MESH is instructed that the file can be deleted since the message has been successfully downloaded.
```